### ISSUE TITLE
Trigger resize, refresh, initialize aren't working

### ISSUE THREAD

**Issue by greguarr** [smashingboxes/OwlCarousel2#842]
Event triggers 'resize.owl.carousel' or 'refresh.owl.carousel' aren't working.

Responsive is working fine when you resize whole window, but when you resize only div container with owl slider - it has still prewious width.
See example:
http://jsfiddle.net/greguarr/x8hs2omz/2/

I try use path, by call resize window event.
It's workind on modern browsers, but I don't know what how it works on others.
There is example with this path:
http://jsfiddle.net/greguarr/x8hs2omz/3/

**Comment by kwayyinfotech** [comment]
I have also same problem

**Comment by DusanMilko** [comment]
"initialized.owl.carousel" is not getting triggered

**Comment by mgntrn** [comment]
Same issue here

**Comment by michaelgiles** [comment]
Same issue here.

**Comment by greg5green** [comment]
I realize your question is about beta 2.4, but this section of code has changed quite a bit since then, so I'm going to speak on beta 3 and where to go from here. There's two problems here from what I can tell:
1. `resize.owl.carousel` isn't a triggerable event -- you can only listen for it.
2. There is no way to manually invalidate a property (and `refresh.owl.carousel` only refreshes things that have been invalidated). This should probably be updated tohave `refresh.owl.carousel` accept a parameter about what should be refreshed.

These updates are something I'd consider important to get done for the next release.

For a more immediate solution in beta 3, you can manually access the `_invalidated` property (with the caveat that this is an internal property, so there are no guarantees about it in the next release). [Working example](http://jsfiddle.net/8jsv3n4o/) or code below:
```
function owlres() {
function owlres() {
     var $carousel = $('.owl-carousel');        
     $carousel.data('owl.carousel')._invalidated.width = true;
     $carousel.trigger('refresh.owl.carousel');
        
    console.log('owlres');
}
```

There's probably something similar you could do for beta 2.4, but the code has changed quite a bit so the above method doesn't work.

**Comment by julianguy** [comment]
I used this approach, set up options and function like so:

```
  var wysiwyg_owloptions = {
	items:1,
	loop: true
  }	
function owlResize($owl) {
	$owl.trigger('destroy.owl.carousel');
	$owl.html($owl.find('.owl-stage-outer').html()).removeClass('owl-loaded');
	$owl.owlCarousel(wysiwyg_owloptions);
}

```

Then just call the function like so:

```
var $owl = $(".carousel).owlCarousel(wysiwyg_owloptions);;
owlResize($owl);
```

**Comment by milanagheda** [comment]
same problem is here tried all solution but nothing is worked

**Comment by camilleb** [comment]
This worked for me : 
`var owl = $('.owl-carousel').data('owl.carousel');owl.onResize();`

**Comment by aarell** [comment]
I'd upgraded to Beta 4 before I noticed a similar problem. I have a product image zoom on Owl images using Magnifik. In zoom mode, if the window is resized (e.g. portrait to landscape phone rotate), Owl seems to resize in the background and gets in a mess. 

Thanks @camilleb - your solution doesn't work for me in beta 4 but I guess the namespace changed in this version (as @greg5green warns about internal changes). I dug down and used owlCarousel namespace instead, so this works for me (in my case wrapped with a small delay to allow settling after the zoom closes):

var owl = $('.owl-carousel').data('owlCarousel');
owl.onResize();

**Comment by greg5green** [comment]
Just to clear this up, that as of today (April 4th, 2016), there is no beta 4. There is a poorly named beta 2.4 and a beta 3.

@aarell you have probably moved backwards to beta 2.4.

**Comment by daviddeutsch** [comment]
Yeah, I should really clear up the versioning in general. I'm happy to take suggestion! @greg5green?

**Comment by sercanbadem** [comment]
Sometimes Owl carousel has calculate problem on resize. If you have position problem, you can set as:
 .owl-carousel{ width:100vw; } i solved with this.

**Comment by vdswouter** [comment]
I know it is not pretty, 
but this worked for me to force a recalculation
```
 this.$carousel.trigger('next.owl.carousel');
 this.$carousel.trigger('prev.owl.carousel');
```

**Comment by rufatZZ** [comment]
@julianguy  Man!. Thanks for your help :)

**Comment by anilljoshi** [comment]
autoHeight is not working in reactjs any Help please?

**Comment by AnthonySaldana** [comment]
I've been dealing with a weird annoying issue for a while where the carousel would not render correctly. Probably because it was rendering before my other scripts processed and causing the sizing calculations to be off.

Anyway, thanks to @greg5green  I was able to just re render it when the time was right!!

This worked for me. Set the invalidated property to true and trigger a refresh!

```
var $carousel = $('.owl-carousel');        
$carousel.data('owl.carousel')._invalidated.width = true;
$carousel.trigger('refresh.owl.carousel');
```

**Comment by heroselohim** [comment]
Ok, none of this solutions worked for me and this is an alternative solution.

Just emit a global 'resize' event.

`window.dispatchEvent(new Event('resize'));`

**Comment by sadikyalcin** [comment]
There are still issues with calculations. Especially when going fullscreen. Refresh / resize does not trigger. Mainly because of the differences in browsers and their transform origins. Only solution is to attach a global resize event like @heroselohim suggested.

**Comment by emreozandac** [comment]
@heroselohim I had an issue very similar to #612  and triggering a resize event solved it. Thanks

**Comment by harshvardhanmalpani** [comment]
@heroselohim I had an issue very similar to #612 and triggering a resize event solved it. Thanks

**Comment by emreozandac** [comment]
Whats this? Whays going on
		
		

		Get Outlook for iOS
	




On Tue, May 1, 2018 at 8:13 PM -0400, "Harshvardhan Malpani" <notifications@github.com> wrote:












@heroselohim I had an issue very similar to #612 and triggering a resize event solved it. Thanks

**Comment by monparaashvin** [comment]
There are issues in Opencart Not Working Owl carousel 
var pdquickview = {
	'PdQwlCarousel : function () {
		var $owl = $('#img-carousel');
		$owl.owlCarousel({
			items: 1,
			singleItem: true,
			margin:10,
			autoPlay: false,
			dots:true
		});
	},
	'ajaxView'	: function (url) {
		$.ajax({
			url 		: url,
			type		: 'get',
			beforeSend	: function() {
				$('#quickview-bg-block').show();
				$('.quickview-load-img').show();
			},
			success		: function(json) {
				$('.quickview-load-img').hide();
				$('#quickview-content').html(json['html']);
				pdquickview.PdQwlCarousel();
				var owl = $('.owl-carousel').data('owl.carousel');
				owl.onResize();
				$(window).trigger('resize');
				tdquickview.appendCloseFrameLink();
				$('#quickview-content').show(600);
			}
		});
	}
};
But not Resize my images on quickview popup.
![untitled-1](https://user-images.githubusercontent.com/11438553/40423948-b2c11ec8-5eb1-11e8-8901-cc31da5a184d.jpg)

**Comment by monparaashvin** [comment]
@kwayyinfotech @michaelgiles @harshvardhanmalpani @mgntrn

**Comment by zodit** [comment]
The carousel is refreshing using the corners of the window, but it is not refreshing using the minimize/maximize button. I am not sure why the resize event is not triggering. I solved following the previous suggestions, adding a little JQuery like this:

` $(window).resize(function(){
$('.owl-carousel').trigger('refresh.owl.carousel');
});`

**Comment by ghost** [comment]
@zodit Thanks, this worked for me. I feel like it should just do that behavior by default.

**Comment by kundankumarmourya** [comment]
I found the owl carousel refresh on window resize. 
https://stackoverflow.com/questions/43371342/reload-owl-carousel-2-when-side-navigation-opens-or-closes

**Comment by oenline** [comment]
> Ok, none of this solutions worked for me and this is an alternative solution.
> 
> Just emit a global 'resize' event.
> 
> `window.dispatchEvent(new Event('resize'));`

This worked for me, thanks a lot.
